### ISSUE TITLE
Imaging barrel reco failing with BryceCanyon 

### ISSUE THREAD

**Issue by DraTeots** [eic/EICrecon#294]
### Environment: 
branch: main

### Steps to reproduce:

When running reco_flags.py

```bash
python3 reco_flags.py  2022-10-27_pgun_e-_epic_brycecanyon_e0.01-30GeV_alldir_1prt_1000evt.edm4hep.root reco_flags_test
```

fails at:

```
Thread model: pthreads
140355769693952: 
       `- eicrecon (0x40f118)
        `- /lib/x86_64-linux-gnu/libpthread.so.0 (0x7fa742785420)
         `- std::vector<dd4hep::DDSegmentation::BitFieldElement, std::allocator<dd4hep::DDSegmentation::BitFieldElement> >::size() const (0x7fa73fef303c)
          `- std::vector<dd4hep::DDSegmentation::BitFieldElement, std::allocator<dd4hep::DDSegmentation::BitFieldElement> >::_M_range_check(unsigned long) const (0x7fa73fef2fc8)
           `- std::vector<dd4hep::DDSegmentation::BitFieldElement, std::allocator<dd4hep::DDSegmentation::BitFieldElement> >::at(unsigned long) const (0x7fa73fef2f84)
            `- dd4hep::DDSegmentation::BitFieldCoder::get(long long, unsigned long) const (0x7fa73fef2aa1)
             `- ImagingPixelReco::execute() (0x7fa73e65cc5e)
              `- CalorimeterHit_factory_EcalBarrelImagingRecHits::Process(std::shared_ptr<JEvent const> const&) (0x7fa73e65bf69)
               `- JFactoryT<edm4eic::CalorimeterHit>::GetOrCreate(std::shared_ptr<JEvent const> const&, JApplication*, int) (0x7fa73e63d6f2)
                `- std::vector<edm4eic::CalorimeterHit const*, std::allocator<edm4eic::CalorimeterHit const*> > JEvent::Get<edm4eic::CalorimeterHit>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const (0x7fa73e642864)
                 `- ProtoCluster_factory_EcalBarrelImagingProtoClusters::Process(std::shared_ptr<JEvent const> const&) (0x7fa73e65f5fe)
                  `- JFactoryT<edm4eic::ProtoCluster>::GetOrCreate(std::shared_ptr<JEvent const> const&, JApplication*, int) (0x7fa73e644b62)
                   `- std::vector<edm4eic::ProtoCluster const*, std::allocator<edm4eic::ProtoCluster const*> > JEvent::Get<edm4eic::ProtoCluster>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const (0x7fa73e648ed4)
                    `- Cluster_factory_EcalBarrelImagingClusters::Process(std::shared_ptr<JEvent const> const&) (0x7fa73e665922)
                     `- JFactoryT<edm4eic::Cluster>::GetOrCreate(std::shared_ptr<JEvent const> const&, JApplication*, int) (0x7fa73f189c22)
                      `- JFactoryT<edm4eic::Cluster>::Create(std::shared_ptr<JEvent const> const&, JApplication*, unsigned long) (0x7fa73e646d3f)
                       `- JEventProcessorPODIO::Process(std::shared_ptr<JEvent const> const&) (0x7fa73c2e9922)
                        `- JEventProcessor::DoMap(std::shared_ptr<JEvent const> const&) (0x7fa741d64a68)
                         `- eicrecon (0x45fe04)
                          `- eicrecon (0x46692e)
                           `- eicrecon (0x468cea)
                            `- eicrecon (0x468c18)
                             `- eicrecon (0x468b77)
                              `- eicrecon (0x468b2e)
                               `- eicrecon (0x468b0e)
                                `- /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x7fa7429d6de4)
                                 `- /lib/x86_64-linux-gnu/libpthread.so.0 (0x7fa742779609)
                                  `- clone (0x7fa74269e133)
```

Full log: 

[imaging_ecal_error.txt](https://github.com/eic/EICrecon/files/9884804/imaging_ecal_error.txt)

CalorimeterHit_factory_EcalBarrelImagingRecHits

**Comment by DraTeots** [comment]
My bad

**Comment by DraTeots** [comment]
No it is persisting problem that is now in main

**Comment by DraTeots** [comment]
https://eicaidata.s3.amazonaws.com/2022-10-27_pgun_e-_epic_arches_e0.01-30GeV_alldir_5prt_2000evt.edm4hep.root

test file

https://eicaidata.s3.amazonaws.com/2022-10-27_pgun_pi+_epic_brycecanyon_e0.01-30GeV_alldir_1prt_1000evt.edm4hep.root

**Comment by DraTeots** [comment]
For arches it just hangs, no error messages

test file
https://eicaidata.s3.amazonaws.com/2022-10-27_pgun_e-_epic_arches_e0.01-30GeV_alldir_5prt_2000evt.edm4hep.root

**Comment by faustus123** [comment]
I'm unable to download the brycecanyon file (error 404 file not found). I am able to download the arches one. It does run for me. It does print a lot of messages like this:

~~~
VolumeManager    ERROR lookupContext: Failed to search Volume context 0017FFFE0100D86F [Unknown identifier]
~~~

It still runs in eic-shell at about 20Hz on ifarm1802. However, using the latest eic-shell with the current main of EICrecon, I only see 4 ReconstructedParticles objects.

Bottom line: I'm not getting the hangs, but I'm also not seeing what I expect for results.

**Comment by faustus123** [comment]
Update: I tried running with my manual build rather than eic-shell and now I am seeing hangs. Investivgating ....

**Comment by faustus123** [comment]
Stack trace is not all that helpful. Seems to be stalled in a call to ```__pthread_once_slow```. Likely due to a deadlock condition. Maybe an exception is being thrown on the first attempt and the call_once mutex is being left in a locked state? Still investigating....

~~~
#0  0x00007f2c8666c1f5 in __pthread_once_slow () from /lib64/libpthread.so.0
(gdb) bt
#0  0x00007f2c8666c1f5 in __pthread_once_slow () from /lib64/libpthread.so.0
#1  0x00007f2c7ee2f3c5 in __gthread_once (__once=0x24e5460, __func=0x7f2c864128c0 <__once_proxy>)
    at /opt/rh/devtoolset-9/root/usr/include/c++/9/x86_64-redhat-linux/bits/gthr-default.h:700
#2  0x00007f2c7ee721ae in std::call_once<void (JFactory::*)(), JFactoryT<edm4eic::CalorimeterHit>*> (__once=..., 
    __f=@0x7f2c61a31f20: &virtual JFactory::Init()) at /opt/rh/devtoolset-9/root/usr/include/c++/9/mutex:683
#3  0x00007f2c7ee61ddb in JFactoryT<edm4eic::CalorimeterHit>::GetOrCreate (this=0x24e53c0, event=warning: RTTI symbol not found for class 'std::_Sp_counted_ptr_inplace<JEvent, std::allocator<JEvent>, (__gnu_cxx::_Lock_policy)2>'
warning: RTTI symbol not found for class 'std::_Sp_counted_ptr_inplace<JEvent, std::allocator<JEvent>, (__gnu_cxx::_Lock_policy)2>'

std::shared_ptr<const JEvent> (use count 2, weak count 1) = {...}, app=0x2410d80, run_number=0)
    at /home/davidl/work/2022.07.11.EICrecon/EICTOP/JANA/v2.0.7/include/JANA/JFactoryT.h:89
#4  0x00007f2c7eeece46 in JFactoryT<edm4eic::CalorimeterHit>::Create (this=0x24e53c0, event=warning: RTTI symbol not found for class 'std::_Sp_counted_ptr_inplace<JEvent, std::allocator<JEvent>, (__gnu_cxx::_Lock_policy)2>'
warning: RTTI symbol not found for class 'std::_Sp_counted_ptr_inplace<JEvent, std::allocator<JEvent>, (__gnu_cxx::_Lock_policy)2>'

std::shared_ptr<const JEvent> (use count 2, weak count 1) = {...}, app=0x2410d80, run_number=0)
    at /home/davidl/work/2022.07.11.EICrecon/EICTOP/JANA/v2.0.7/include/JANA/JFactoryT.h:132
#5  0x00007f2c7cd20488 in JEventProcessorPODIO::Process (this=0x24dcd20, event=warning: RTTI symbol not found for class 'std::_Sp_counted_ptr_inplace<JEvent, std::allocator<JEvent>, (__gnu_cxx::_Lock_policy)2>'
warning: RTTI symbol not found for class 'std::_Sp_counted_ptr_inplace<JEvent, std::allocator<JEvent>, (__gnu_cxx::_Lock_policy)2>'

std::shared_ptr<const JEvent> (use count 2, weak count 1) = {...})
    at /home/davidl/work/2022.09.02.EICrecon/EICrecon/src/services/io/podio/JEventProcessorPODIO.cc:281
#6  0x00007f2c8233e781 in JEventProcessor::DoMap(std::shared_ptr<JEvent const> const&) ()
   from /home/davidl/work/2022.09.02.EICrecon/EICrecon/lib/EICrecon/plugins/log.so
#7  0x000000000045a083 in JEventProcessorArrow::execute(JArrowMetrics&, unsigned long) ()
#8  0x0000000000460090 in JWorker::loop() ()
#9  0x0000000000462028 in void std::__invoke_impl<void, void (JWorker::*)(), JWorker*>(std::__invoke_memfun_deref, void (JWorker::*&&)(), JWorker*&&) ()
#10 0x0000000000461f6a in std::__invoke_result<void (JWorker::*)(), JWorker*>::type std::__invoke<void (JWorker::*)(), JWorker*>(void (JWorker::*&&)(), JWorker*&&) ()
#11 0x0000000000461ed9 in void std::thread::_Invoker<std::tuple<void (JWorker::*)(), JWorker*> >::_M_invoke<0ul, 1ul>(std::_Index_tuple<0ul, 1ul>) ()
#12 0x0000000000461e94 in std::thread::_Invoker<std::tuple<void (JWorker::*)(), JWorker*> >::operator()() ()
#13 0x0000000000461e78 in std::thread::_State_impl<std::thread::_Invoker<std::tuple<void (JWorker::*)(), JWorker*> > >::_M_run() ()
#14 0x000000000049a3b0 in execute_native_thread_routine ()
#15 0x00007f2c8666dea5 in start_thread () from /lib64/libpthread.so.0
#16 0x00007f2c85b76b0d in clone () from /lib64/libc.so.6
~~~

**Comment by DraTeots** [comment]
Should we close it now then?

**Comment by faustus123** [comment]
Yes, this was fixed in PR#305.
